mods/env_sounds/init.lua

```diff
@@ -0,0 +1,121 @@
+-- Parameters
+
+local radius = 8 -- Water node search radius around player
+
+-- End of parameters
+
+
+local handles = {}
+local river_source_sounds = minetest.settings:get_bool("river_source_sounds")
+
+
+-- Update sound for player
+
+local function update_sound(player)
+	local player_name = player:get_player_name()
+	-- Search for water nodes in radius around player
+	local ppos = player:getpos()
+	local areamin = vector.subtract(ppos, radius)
+	local areamax = vector.add(ppos, radius)
+	local wpos, num
+	if river_source_sounds then
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_source",
+				"default:river_water_flowing"
+			}
+		)
+	else
+		wpos, num = minetest.find_nodes_in_area(
+			areamin,
+			areamax,
+			{
+				"default:water_flowing",
+				"default:river_water_flowing"
+			}
+		)
+	end
+	-- Total number of waters in radius
+	local waters = (num["default:water_flowing"] or 0) +
+		(num["default:river_water_source"] or 0) +
+		(num["default:river_water_flowing"] or 0)
```

Comment:
Why not `local waters = #wpos`?

Reply:
Hmm good point, i'm not sure why i did it this way.

Reply:
I'll test the values to check they're identical (i expect they are) and simplify.

Reply:
Indeed they are identical.